internal/api/grpc/server/middleware/execution_interceptor.go

```diff
@@ -0,0 +1,127 @@
+package middleware
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/grpc"
+
+	"github.com/zitadel/zitadel/internal/api/authz"
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/execution"
+	"github.com/zitadel/zitadel/internal/query"
+	exec_repo "github.com/zitadel/zitadel/internal/repository/execution"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+func ExecutionHandler(queries *query.Queries) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		request, err := executeTargetsForRequest(ctx, queries, info.FullMethod, req)
+		if err != nil {
+			return nil, err
+		}
+
+		resp, err := handler(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+
+		return executeTargetsForResponse(ctx, queries, info.FullMethod, req, resp)
+	}
+}
+
+func executeTargetsForRequest(ctx context.Context, queries ExecutionQueries, fullMethod string, req interface{}) (interface{}, error) {
+	targets, err := queryTargets(ctx, queries, fullMethod, domain.ExecutionTypeRequest)
+	if err != nil {
+		// if no targets are found, return without any calls
+		if zerrors.IsNotFound(err) {
+			return req, nil
+		}
+		return nil, err
+	}
+
+	ctxData := authz.GetCtxData(ctx)
+	info := &execution.ContextInfoRequest{
+		FullMethod: fullMethod,
+		InstanceID: authz.GetInstance(ctx).InstanceID(),
+		ProjectID:  ctxData.ProjectID,
+		OrgID:      ctxData.OrgID,
+		UserID:     ctxData.UserID,
+		Request:    req,
+	}
+
+	request, err := execution.CallTargetsRequest(ctx, targets, info)
+	if err != nil {
+		// if an error is returned still return also the original request
+		return req, err
+	}
+	return request, err
```

Comment:
Why is this not logic of the called function?

Reply:
As this is error handling on the interceptor part, which should not be included in how the call is handled.